Assets/HoloToolkit/Input/Scripts/GazeStabilizer.cs

```diff
@@ -105,10 +105,10 @@ private void AddGazeSample(Vector3 positionSample, Vector3 directionSample)
                 // Remove from front if we exceed stored samples.
                 if (stabilitySamples.Count >= StoredStabilitySamples)
                 {
-                    stabilitySamples.Dequeue();
+                    stabilitySamples.RemoveAt(0);
```

Comment:
Generally I think this is closer to the right implementation, but now you've introduced a block copy here. My advice is use implement a wrapping write index to avoid this new downside.